detector/other/FieldMapBrBz.cpp

```diff
@@ -150,44 +157,156 @@ void FieldMapBrBz::fillFieldMapFromTree(const std::string& filename, const std::
 
   if( strs.size() != 5 ) {
     std::stringstream error;
-    error << "FieldMap[ERROR]: the treeDescription " << treeString
+    error << "FieldMapBrBz[ERROR]: the treeDescription " << treeString
 	  << " is not complete. For example 'fieldmap:rho:z:Brho:Bz'";
     throw std::runtime_error( error.str() );
   }
 
+  //Getting the tree name and branch variables names
```

Comment:
You're right. Now you cannot define any tree variables names (coordinated and Bfield components), but you don't have to specify any given order. I did it this way because people usually use the same variables naming.

So, we can come back to the previous convention in which the names ordering is what is important, but this has the possibility of error by the user in specifying a wrong order. In this case the code can still work but with swapping of tree variables meaning.

Up to you to decide what is best.